lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -236,8 +236,8 @@ def get_diff_image(self):
                 self._last_renderer, renderer)
             self._force_full = False
             self._png_is_old = False
-
-        return buff
+            self.buff = buff
+        return self.buff
```

Comment:
Would it be better to return `buff` inside the if block and return `None` if there is no changes?